docs/REST API/rest-api.md

```diff
@@ -3630,6 +3643,79 @@ Retrieves a list of the configuration profiles assigned to a host.
 }
 ```
 
+### Lock host
+
+_Available in Fleet Premium_
+
+Sends a command to lock the specified macOS, Linux, or Windows host. The host is locked once it comes online.
+
+`GET /api/v1/fleet/hosts/:id/lock`
+
+#### Parameters
+
+| Name       | Type              | In   | Description                                                                   |
+| ---------- | ----------------- | ---- | ----------------------------------------------------------------------------- |
+| id | integer | path | **Required**. ID of the host to be locked. |
+
+#### Example
+
+`GET /api/v1/fleet/hosts/123/lock`
+
+##### Default response
+
+`Status: 200`
+
+Requests to lock macOS hosts will also include the following response data:
+```json
+{
+  "host_id": 8,
+  "unlock_pin": 123456
+}
+```
+
+### Unlock host
+
+_Available in Fleet Premium_
+
+Sends a command to unlock the specified Windows or Linux host. The host is unlocked once it comes online.
+
+`GET /api/v1/fleet/hosts/:id/unlock`
+
+#### Parameters
+
+| Name       | Type              | In   | Description                                                                   |
+| ---------- | ----------------- | ---- | ----------------------------------------------------------------------------- |
+| id | integer | path | **Required**. ID of the host to be unlocked. |
+
+#### Example
+
+`GET /api/v1/fleet/hosts/:id/unlock`
+
+##### Default response
+
+`Status: 200`
+
```

Comment:
We should add the example error if it's a mac, unlock won't work for mac

Reply:
@noahtalerman new plan is to have this endpoint track "viewed PIN" activity and return the unlock PIN for macOS (instead of erroring)?